crates/oxc_linter/src/rules/vue/require_typed_ref.rs

```diff
@@ -0,0 +1,317 @@
+use oxc_ast::{AstKind, ast::Argument};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{AstNode, context::LintContext, rule::Rule};
+
+fn require_typed_ref_diagnostic(span: Span, name: &str) -> OxcDiagnostic {
+    let msg = format!(
+        "Specify type parameter for `{name}` function, otherwise created variable will not be typechecked."
+    );
+    OxcDiagnostic::warn(msg)
+        .with_help("Provide an explicit type parameter or an initial value.")
+        .with_label(span)
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct RequireTypedRef;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Require `ref` and `shallowRef` functions to be strongly typed.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// With TypeScript it is easy to prevent usage of any by using noImplicitAny.
```

Comment:
NIT: put source code text in inside ``, like `noImplicitAny`, `ref()`, etc...
https://eslint.vuejs.org/rules/require-typed-ref.html